### PR TITLE
Describe java OTLP exporter options

### LINKED ISSUE #3271: The java story around the OTLP exporter
The java story around the OTLP exporter is actually pretty nuanced... What you describe here works, but its not the only way.

There are two versions of OTLP we support: `grpc` and `http/protobuf`

These have exporter classes of the form `OtlpGrpc{Signal}Exporter` for `grpc`, and `OtlpHttp{Signal}Exporter` for `http/protobuf`.

Under the hood, there are multiple sender implementations for each of these. Sender is an abstraction we created to be able to plugin different grpc / http client implementations to fulfill the OTLP contract. You select which sender implementation you want by optionally include / excluding dependencies. Here is the list (as of the 8/11 1.29.0 release):

- `opentelemetry-exporter-sender-okhttp` - The default, included automatically with `opentelemetry-exporter-otlp`. This includes an okhttp based sender for both `OtlpGrpc{Signal}Exporter` and `OtlpHttp{Signal}Exporter`. It works for many people, and is included in the otel java agent distribution. However, certain use cases want to avoid the okhttp dependency, since it has a transitive dependency on kotlin. The alternatives below accommodate these use cases.
- `opentelemetry-exporter-sender-jdk` - Provides an jdk 11+ http client sender implementation for `OtlpHttp{Signal}Exporter`. You can use this if you have java 11+, and prefer to avoid extra dependencies since this uses code that comes built into java. If using, you'll want to explicitly exclude the default `opentelemetry-exporter-sender-okhttp`, and note that you wont be able to use this with `OtlpGrpc{Signal}Exporter`.
- `opentelemetry-exporter-sender-grpc-managed-channel` - Provides a sender implementation for `OtlpGrpc{Signal}Exporter` based on the [grpc java client](https://github.com/grpc/grpc-java). This supports folks who insist on using `ManagedChannel`.

Although its a bit complex, there are valid use cases that demand each of these solutions. We think that between these solutions we can make virtually everyone happy. 

Eventually, i do want to document what I've written down here ([see here](https://github.com/open-telemetry/opentelemetry-java/pull/5617#issuecomment-1664755986)). Not sure whether we should include it in this PR or a followup.

_Originally posted by @jack-berg in https://github.com/open-telemetry/opentelemetry.io/pull/3144#discussion_r1287632380_

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #3271.

### CODE REVIEW

File: content/en/docs/instrumentation/java/exporters.md [hunk 1]
```diff
@@ -45,6 +50,46 @@ dependencies {
...
+- `grpc` - gRPC implementation of OTLP exporters, represented by
+  `OtlpGrpcSpanExporter`, `OtlpGrpcMetricExporter`, `OtlpGrpcLogRecordExporter`.
+- `http/protobuf` - HTTP with protobuf encoded payload implementation of OTLP
+  exporters, represented by `OtlpHttpSpanExporter`, `OtlpHttpMetricExporter`,
+  `OtlpHttpLogRecordExporter`.
+
+A sender is an abstraction which allows different gRPC / HTTP client
+implementations to fulfill the OTLP contract. Regardless of the sender
+implementation, the same exporter classes are used. A sender implementations is
+automatically used when it is detected on the classpath. The sender
```
Comment (svrnm) [inline content/en/docs/instrumentation/java/exporters.md:65]: add `classpath` to the ignore list in line 4

We can later consider to have it added to the global list as well.

File: content/en/docs/instrumentation/java/exporters.md [hunk 2]
```diff
@@ -45,6 +50,46 @@ dependencies {
...
+implementations are described in detail below:
+
+- `{groupId}:{artifactId}` - Sender description.
+- `io.opentelemetry:opentelemetry-exporter-sender-okhttp` - The default sender,
```
Comment (svrnm) [inline content/en/docs/instrumentation/java/exporters.md:69]: Same here, add `okhttp` to line 4 to ignore it

File: content/en/docs/instrumentation/java/exporters.md [hunk 3]
```diff
@@ -45,6 +50,46 @@ dependencies {
...
+  OpenTelemetry Java agent. This includes an
+  [OkHttp](https://square.github.io/okhttp/) based implementation for both the
+  `grpc` and `http/protobuf` versions of the protocol, and will be suitable for
+  most users. However, OkHttp has a transitive dependency on kotlin which is
+  problematic in some environments.
+- `io.opentelemetry:opentelemetry-exporter-sender-jdk` - This sender includes a
+  JDK 11+
+  [HttpClient](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html)
+  based implementation for the `http/protobuf` version of the protocol. It
+  requires zero additional dependencies, but requires java 11+. To use, include
```
Comment (svrnm) [inline content/en/docs/instrumentation/java/exporters.md:80]:
```suggestion
  requires zero additional dependencies, but requires Java 11+. To use, include
```

File: content/en/docs/instrumentation/java/exporters.md [hunk 4]
```diff
@@ -45,6 +50,46 @@ dependencies {
...
+  the artifact and explicitly exclude the default
+  `io.opentelemetry:opentelemetry-exporter-sender-okhttp` dependency.
+- `io.opentelemetry:opentelemetry-exporter-sender-grpc-managed-channel` - This
+  sender includes a [grpc-java](https://github.com/grpc/grpc-java) based
+  implementation for the `grpc` version of the protocol. To use, include the
+  artifact, explicitly exclude the default
+  `io.opentelemetry:opentelemetry-exporter-sender-okhttp` dependency, and
+  include one of the
+  [grpc transport implementations](https://github.com/grpc/grpc-java#transport).
```
Comment (svrnm) [inline content/en/docs/instrumentation/java/exporters.md:89]:
```suggestion
  [gRPC transport implementations](https://github.com/grpc/grpc-java#transport).
```

**Review by svrnm — CHANGES_REQUESTED** [review]
lgtm overall, some minor points regarding cspell & text-lint that need to be fixed

File: content/en/docs/instrumentation/java/exporters.md [hunk 5]
```diff
@@ -45,6 +50,46 @@ dependencies {
...
+
+- `grpc` - gRPC implementation of OTLP exporters, represented by
+  `OtlpGrpcSpanExporter`, `OtlpGrpcMetricExporter`, `OtlpGrpcLogRecordExporter`.
+- `http/protobuf` - HTTP with protobuf encoded payload implementation of OTLP
+  exporters, represented by `OtlpHttpSpanExporter`, `OtlpHttpMetricExporter`,
+  `OtlpHttpLogRecordExporter`.
+
+A sender is an abstraction which allows different gRPC / HTTP client
+implementations to fulfill the OTLP contract. Regardless of the sender
+implementation, the same exporter classes are used. A sender implementations is
```
Comment (cartermp) [inline content/en/docs/instrumentation/java/exporters.md:64]:
```suggestion
implementation, the same exporter classes are used. A sender implementation is
```

File: content/en/docs/instrumentation/java/exporters.md [hunk 6]
```diff
@@ -45,6 +50,46 @@ dependencies {
...
+  the artifact and explicitly exclude the default
+  `io.opentelemetry:opentelemetry-exporter-sender-okhttp` dependency.
+- `io.opentelemetry:opentelemetry-exporter-sender-grpc-managed-channel` - This
+  sender includes a [grpc-java](https://github.com/grpc/grpc-java) based
+  implementation for the `grpc` version of the protocol. To use, include the
+  artifact, explicitly exclude the default
+  `io.opentelemetry:opentelemetry-exporter-sender-okhttp` dependency, and
+  include one of the
+  [grpc transport implementations](https://github.com/grpc/grpc-java#transport).
+
```
Comment (cartermp) [inline content/en/docs/instrumentation/java/exporters.md:90]: I think what's missing is an explanation of when I'd want to use a different sender implementation than the default.

Comment (cartermp) [reply content/en/docs/instrumentation/java/exporters.md:90]: Gotcha. I suppose most java developers who are in this situation will recognize when they need to swap a sender implementation.

**Review by cartermp — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
thanks @jack-berg :-)

**Pull request merged**